tango/step.py

```diff
@@ -176,6 +175,13 @@ def __init__(
 
         self._config = step_config
 
+    @property
+    def logger(self) -> logging.Logger:
+        """
+        A :class:`logging.Logger` that can be used within the :meth:`run()` method.
+        """
+        return logging.getLogger(self.__class__.__name__)
```

Comment:
Why is it better to have this as a property instead of as a member?

Reply:
When you set it as a member, the logger will get initialized before `initialize_logging()` is called. I was worried this would cause issues. But maybe it's okay.